### DISCUSSION TITLE
help: failed to decompress rpm files with rpm2cpio&cpio

### DISCUSSION THREAD

**Discussion by zhuizhuhaomeng** [rpm-software-management/rpm#2628] [Q&A] [answered]
I want to decompress the kernel-5.14.0-284.18.1.el9_2.aarch64.rpm with the following.
I can not decompress successfully.

```shell
rpm2cpio kernel-5.14.0-284.18.1.el9_2.aarch64.rpm | cpio -div
1 block
```

Then I ran `rpm2cpio kernel-5.14.0-284.18.1.el9_2.aarch64.rpm > a.cpio`.
I found the a.cpio only has 124 bytes.

What's the correct command to decompress the rpm file ?

**Comment by zhuizhuhaomeng** [comment]
I think the kernel-5.14.0-284.18.1.el9_2.aarch64.rpm contains no file.
But why the file is 3.5M.  what is the content inside it?

**Comment by pmatilai** [comment, marked as answer]
rpm2cpio doesn't check for any rpmlib() dependencies the package may have, try 'rpm -i' on the src.rpm for that. Most likely the kernel package you're looking at is compressed with an algorithm (zstd) that the rpm you're using doesn't support.
